cmd/completion/main.go

```diff
@@ -16,52 +19,94 @@ const (
 	registrySecretsDir = "/var/build-secrets"
 	reportFilePath     = "/var/report/report.toml"
 	notarySecretDir    = "/var/notary/v1"
+	buildNumberKey     = "buildNumber"
+	buildTimestampKey  = "buildTimestamp"
 )
 
 var (
+	buildNumber             string
+	buildTimestamp          string
 	notaryV1URL             string
 	dockerCredentials       flaghelpers.CredentialsFlags
 	dockerCfgCredentials    flaghelpers.CredentialsFlags
 	dockerConfigCredentials flaghelpers.CredentialsFlags
+	cosignAnnotations       flaghelpers.CredentialsFlags
 	logger                  *log.Logger
 )
 
 func init() {
+	flag.StringVar(&buildNumber, "build-number", "1", "Build number")
+	flag.StringVar(&buildTimestamp, "build-timestamp", "", "Build timestamp")
 	flag.StringVar(&notaryV1URL, "notary-v1-url", "", "Notary V1 server url")
 	flag.Var(&dockerCredentials, "basic-docker", "Basic authentication for docker of the form 'secretname=git.domain.com'")
 	flag.Var(&dockerCfgCredentials, "dockercfg", "Docker Cfg credentials in the form of the path to the credential")
 	flag.Var(&dockerConfigCredentials, "dockerconfig", "Docker Config JSON credentials in the form of the path to the credential")
-
+	flag.Var(&cosignAnnotations, "cosign-annotations", "Cosign custom signing annotations")
 	logger = log.New(os.Stdout, "", 0)
 }
 
 func main() {
 	flag.Parse()
 
-	if notaryV1URL != "" {
-		creds, err := dockercreds.ParseMountedAnnotatedSecrets(registrySecretsDir, dockerCredentials)
+	creds, err := dockercreds.ParseMountedAnnotatedSecrets(registrySecretsDir, dockerCredentials)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	for _, c := range append(dockerCfgCredentials, dockerConfigCredentials...) {
```

Comment:
I realize this was the existing behavior in completion but, it seems unnecessary to rebuild the keychain to just write to  `docker/config.json` again. Can we just mount the same credentials from export into completion?

Reply:
Did you mean that I remount the `homeVolume` and set the Env `HOME` to `/builder/home`?

I noticed that `build-init` would save the config to `/builder/home/docker/config.json` and could therefore be reused by completion during the regular `buildPod` flow. However, during the `rebasePod` flow, as the only initContainer is the "rebase" container. It does load some docker credentials but it does not save it. I could introduce the "Save" portion to the rebase image and therefore the completion container could leverage a shared volume for both buildPod and rebasePod scenarios. Is it ideal to introduce this logic into "rebase" when it itself doesn't need the file to be saved just so we can leverage the byproduct of that outside of that container in completion?

The keychain still needed to be built or loaded into a variable to be passed into Notary while cosign only requires the config to be writing to a file named `config.json`. The keychain could be built from the `config.json` though instead of from each secret individually if mounting a shared volume is the way to go

Reply:
@matthewmcnew

Reply:
I forgot about the complexities introduced by the rebase container. I think it would eventually be a good idea for the rebase container to "save" the credentials to a volume that can be loaded in completion. I don't think you need to introduce this in your PR.

Reply:
Since that is the case, I will leave the current credential generation and saving inside the completion image. When we introduce and find a more uniform way of saving credentials, we can adjust the completion container then. For cosign portion in the completion container, with the shared volume during non rebase case, it seemed to work from my quick testing when mounting the shared home volume. For the notary portion, it is still likely that the keychain would need to be loaded to be passed into the notary sign function. Is there a place I can put stories or track this concern? Do I just make a Github issue saying to standardize docker credential saving using shared volumes to be more consistent between rebasePod/buildPod/initContainers/containers?